comments/data_ingest_gotchas.md

```diff
@@ -0,0 +1,68 @@
+
+**forLayer values must adhere this format!**. If you try and pass something in a different format like `level-1` then the client won't parse it and your labels won't display at all. 
+
+## Linking labels to stakeholder data from Excel workbooks
+
+Consider a map that displays a set of stakeholder entities. The stakeholders and their attributes -- name, description, any relationships -- are defined in an Excel workbook and converted to RDF triples via a transform. However, the Excel workbook does not contain any information on how and where to display the stakeholders on the map. This information is provided by the labels export from Illustrator, which comes in the form of a .json file which is converted to triples by the `labels.py` script in `map-tools`.
+
+We now have two sets of triples from the labels and the workbook that, when combined, would give us all of the data we need to display the stakeholders properly. We just need to link the two sets of data together so that the model knows they're referring to the same thing.
+
+The default behaviour for the `labels.py` script in `map-tools` is to attach a `forThing` attribute to each labels that points at another IRI. The client uses this to attach labels to other objects in the data model with the `forThing` IRI. The format of the IRI is as follows:
+
+    'vm:_thing_{row[qcontents].as_slug}'
+
+`row[qcontents]` is the name of the stakeholder as represented on the label in Illustrator, and should match the name of the stakeholder in the workbook. The problem is that whatever transform is converting the workbook data into triples is probably formatting the stakeholder IRI like this:
+
+    'vm:_stakeholder_{row[Category / Stakeholder name].as_slug}'
+
+They're typed differently -- the labels IRI has the `_thing` prefix, while the workbook IRI has the `_stakeholder` prefix. The workbook version is more correct, but the labels has to use a more generic prefix because labels aren't just attached to stakeholders, they can go on painpoints, categories or just about anything.
+
+So we need to create a linking reference that connects our `_thing`-typed IRI to our `_stakeholder`-typed one. This is done by inserting an extra triple into the workbook transform:
+
+    'lets': {
+        'iri': 'vm:_stakeholder_{row[Category / Stakeholder name].as_slug}',
+    },
+    'triples': [
+        ...
+        ('{iri}', 'owl:sameAs',
+            'vm:_thing_{row[Category / Stakeholder name].as_slug}'),
+    ],
+
+This `owl:sameAs` triple is saying that the IRI `vm:_stakeholder_<stakeholder_name>` from the workbook is the same thing as the `vm:_thing_<stakeholder_name>` IRI from the labels data. When the client follows the `forThing` reference to the object the labels are supposed to be attached to, it will now find the data under `vm:_stakeholder` as it is supposed to.

```

Comment:
The last bit here is the transform needs the `--no-resolve-same` argument if part of a multistage transform with the labels last - as this defined alias only wants to be resolved right at the end after the labels have been included.